src/app/users/request-phone-code.ts

```diff
@@ -22,16 +23,19 @@ export const requestPhoneCodeWithCaptcha = async ({
   logger,
   ip,
 }: {
-  phone: PhoneNumber
+  phone: string
   geetest: GeetestType
   geetestChallenge: string
   geetestValidate: string
   geetestSeccode: string
   logger: Logger
   ip: IpAddress
-}): Promise<true | CaptchaError | UnknownPhoneProviderServiceError> => {
+}) => {
```

Comment:
type is automatically being detected by Typescript

Reply:
But if we don't set it explicitly (ie. communicate to the client what the return type should be) and we change the implementation to return something else (perhaps by accident) then this can cause unintended errors right?

In other words if the return is inferred and the implementation changes to return a different type then we have broken the contract inadvertently.

Reply:
should be `| ApplicationError`